foreign_cc/cmake.bzl

```diff
@@ -19,24 +19,21 @@ load(
     "cc_external_rule_impl",
     "create_attrs",
 )
-load(
-    "//toolchains/native_tools:tool_access.bzl",
-    "get_cmake_data",
-    "get_make_data",
-    "get_ninja_data",
-)
+load("//toolchains/native_tools:tool_access.bzl", "get_cmake_data", "get_ninja_data")
 
 def _cmake_impl(ctx):
     cmake_data = get_cmake_data(ctx)
-    make_data = get_make_data(ctx)
+    ninja_path = None
 
-    tools_deps = ctx.attr.tools_deps + cmake_data.deps + make_data.deps
+    tools_deps = ctx.attr.tools_deps + cmake_data.deps
 
-    ninja_data = get_ninja_data(ctx)
-    make_commands = ctx.attr.make_commands
-    if _uses_ninja(ctx.attr.make_commands):
-        tools_deps += ninja_data.deps
-        make_commands = [command.replace("ninja", ninja_data.path) for command in make_commands]
+    # For generators using ninja, make sure it's available
+    for option in ctx.attr.cmake_options:
+        if "ninja" in option.lower():
```

Comment:
`cmake --help` on macOS gives the following list of generators:
```
* Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Xcode                        = Generate Xcode project files.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.
                               ```
It seems that all ninja generators use the capitalisation `Ninja` so I think it's stronger to search for that as searching all of cmake_options seems like it could hit this due to matching paths or config options. Ultimately though I guess the only impact this may have is a bit of extra disk space utilisation and some extra inputs to the cache key for this action?

Reply:
I've since revised how generators are detected. Good call on this one.